frontend/lib/models/transaction.dart

```diff
@@ -1,145 +1,48 @@
+import 'package:economicalc_client/models/receipt.dart';
+import 'package:economicalc_client/models/bank_transaction.dart';
+import 'package:intl/intl.dart';
+
 class Transaction {
-  late String id;
-  late String accountId;
-  late Amount amount;
-  late Descriptions descriptions;
-  late Dates dates;
-  late Types types;
-  late String status;
-  late String providerMutability;
+  int? id;
+  DateTime date;
+  double? totalAmount;
+  String? store;
+  int? bankTransactionID;
+  int? receiptID;
+  int? categoryID;
```

Comment:
Why not save an object of type category here?

Reply:
Same reasoning as for BankTransaction. The reason we still have categoryDesc is that it fixes an issue with the category dropdown that I wasn't keen on looking into right now.